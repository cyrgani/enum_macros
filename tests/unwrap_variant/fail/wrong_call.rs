use enum_macros::UnwrapVariant;

#[derive(UnwrapVariant)]
enum Test {
    #[unwrap(ref)]
    A(String),
    B(usize),
}

fn main() {
    let _a = Test::unwrap_a_ref(1);
}
