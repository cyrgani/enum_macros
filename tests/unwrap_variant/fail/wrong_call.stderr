error[E0308]: mismatched types
  --> tests/unwrap_variant/fail/wrong_call.rs:11:33
   |
11 |     let _a = Test::unwrap_a_ref(1);
   |              ------------------ ^ expected `&Test`, found integer
   |              |
   |              arguments to this function are incorrect
   |
note: method defined here
  --> tests/unwrap_variant/fail/wrong_call.rs:3:10
   |
3  | #[derive(UnwrapVariant)]
   |          ^^^^^^^^^^^^^
   = note: this error originates in the derive macro `UnwrapVariant` (in Nightly builds, run with -Z macro-backtrace for more info)
